homeassistant/components/boschshc/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for Bosch Smart Home Controller integration."""
+import logging
+
+from boschshcpy import SHCSession
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_ICON, CONF_IP_ADDRESS, CONF_NAME
+
+from .const import CONF_SSL_CERTIFICATE, CONF_SSL_KEY
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Bosch SHC"): str,
+        vol.Required(CONF_IP_ADDRESS): str,
+        vol.Required(CONF_SSL_CERTIFICATE): str,
+        vol.Required(CONF_SSL_KEY): str,
+        vol.Optional(CONF_ICON): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    session = await hass.async_add_executor_job(
+        SHCSession,
+        data[CONF_IP_ADDRESS],
+        data[CONF_SSL_CERTIFICATE],
+        data[CONF_SSL_KEY],
+        False,
+    )
+    status = session.information.version
+
+    if status == "n/a":
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": data[CONF_NAME]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Bosch SHC."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                # Check if already configured
+                await self.async_set_unique_id(user_input[CONF_IP_ADDRESS])
```

Comment:
Ip address is not a valid unique id as it can change. The unique id must both be unique and stable.

If there's no valid unique id, we can just check if the ip address is in existing config entries for this domain and abort if so.